sys/net/application_layer/gcoap/gcoap.c

```diff
@@ -570,6 +570,32 @@ static void _find_obs_memo_resource(gcoap_observe_memo_t **memo,
 }
 
 /*
+ * Check listener structure
+ *
+ *  - resources should be sorted alphabetically
+ */
+#ifdef DEVELHELP
+static void _check_listener(gcoap_listener_t *listener)
```

Comment:
IMHO a check function should have return value, i.e., `0` on success and anything else on error.